frontend/src/concepts/pipelines/content/createRun/contentSections/RunTypeSectionScheduled.tsx

```diff
@@ -39,7 +40,8 @@ const RunTypeSectionScheduled: React.FC<RunTypeSectionScheduledProps> = ({ data,
         onChange={(end) => onChange({ ...data, end })}
         adjustNow={(now) => {
           if (data.start) {
-            const start = new Date(`${data.start.date} ${data.start.time}`);
+            const convertedTime = convertToTwentyFourHourTime(data.start.time);
+            const start = new Date(`${data.start.date}T${convertedTime}:00.000`);
```

Comment:
I'd like to see this conversion moved to a function and reused elsewhere.
Something like this:
```
export const toDate = ({ date, time }: RunDateTime) =>
  new Date(`${date}T${convertToTwentyFourHourTime(time)}:00.000`);
```